core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2421,6 +2396,7 @@ class Log(@volatile private var _dir: File,
         newSegments.foreach { splitSegment =>
           splitSegment.close()
           splitSegment.deleteIfExists()
+          producerStateManager.deleteIfExists(splitSegment.baseOffset)
```

Comment:
It doesn't seem that we generate producer snapshot files for those new segments?